### PR TITLE
doc: add parameter update testing to developer guide

### PR DESCRIPTION
**Pull request by jimthompson5802**
This is the update to the Developer Guide that describe how to implement the parameter update test procedure  into the unit tests.

This PR should be merged after all of these code related PRs are merged:
* https://github.com/ludwig-ai/ludwig/pull/2194
* https://github.com/ludwig-ai/ludwig/pull/2252
* https://github.com/ludwig-ai/ludwig/pull/2332
* https://github.com/ludwig-ai/ludwig/pull/2354

### CODE REVIEW

File: docs/developer_guide/unit_test_design_guidelines.md [hunk 1]
```diff
@@ -208,3 +208,178 @@ RANDOM_SEED = 1919
 torch.manual_seed(RANDOM_SEED)
 input_tensor = torch.randn([BATCH_SIZE, input_size], dtype=torch.float32)
 ```
+
+## Test for parameter updates
+
+The utility function `check_module_parameters_updated()` in the `tests.integration_tests.parameter_utils` module
+is available to test whether Ludwig modules, e.,g encoders, combiners, decoders and related sub-components are updating
```
Comment (arnavgarg1) [inline docs/developer_guide/unit_test_design_guidelines.md:215]: nit: change e.,g to e.g.

**Comment by arnavgarg1** [comment]
@jimthompson5802 Looks great! Thanks for a very clear and easy-to-follow write-up. 

I've updated the PR to merge into the `0.6` branch instead of `master` since the parameter testing changes that you've added to Ludwig will most likely make it Ludwig 0.6 instead of a new branch cut of 0.5.x. 

cc: @justinxzhao

**Review by arnavgarg1 — APPROVED** [review]

File: docs/developer_guide/unit_test_design_guidelines.md (hunk 1 above)
Comment (jimthompson5802) [reply docs/developer_guide/unit_test_design_guidelines.md]: corrected.

**Comment by jimthompson5802** [comment]
@arnavgarg1 Got it...`0.6` branch.  

thank you for the feedback.

**Review by justinxzhao — APPROVED** [review]

**Comment by arnavgarg1** [comment]
@jimthompson5802 Is this good to merge?

**Pull request merged**